pkg/rtc/types/interfaces.go

```diff
@@ -214,10 +200,22 @@ type PublishedTrack interface {
 
 	// callbacks
 	AddOnClose(func())
+}
+
+//counterfeiter:generate . LocalPublishedTrack
+type LocalPublishedTrack interface {
+	PublishedTrack
```

Comment:
I'm a bit confused with the 4 interfaces for media track.
* PublishedTrack
* MediaTrack
* LocalMediaTrack
* LocalPublishedTrack

how do they differ? is it sufficient to just have MediaTrack and LocalMediaTrack instead?

Reply:
`PublishedTrack` and `LocalPublishedTrack` are pretty from the publisher side. Things like `GetAudioLevel`, `GetConnectionQuality` live there and it is also used to upwards to room/roommanager callbacks.

`MediaTrack` is the interface used by downstream thing, i. e. SubscribedTrack uses the MediaTrack interface.

`LocalMediaTrack` should actually be removed and be put in `LocalPublishedTrack` as that is how it is used (i. e. get max audio loss from remote and it goes via room/participant -> track. Actually the max subscribed quality from other nodes also go through the path of room/participant -> track. So, that should also technically be moved from `MediaTrack` to `PublishedTrack`.

I am fine making just two `MediaTrack` and `LocalMediaTrack` (any way these are tracks published into the system and it can have subscribers) and use it consistently.

If you are okay with that, I can try merging `PublishedTrack` into `MediaTrack` and let that be the one true one.